lexicon/providers/safedns.py

```diff
@@ -0,0 +1,200 @@
+                record for record in records
+                if record['type'] == rtype]
+        if name:
+            records = [
+                record for record in records
+                # DNS should not be case-sensitive, so perform lower-case comparison
+                if record['name'].lower() == self._full_name(name).lower()]
+        if content:
+            records = [
+                record for record in records
+                if record['content'].lower() == content.lower()]
+
+        LOGGER.debug('list_records: %s', records)
+        return records
+
+    def _create_record(self, rtype, name, content):
+
+        # Check whether the record already exists with the same  rtype, name & content.
+        # If so, claim to have added the record, but dont't do anything.
+        records = self._list_records(rtype, name, content)
+        if records:
+            LOGGER.debug('create_record: (ignored, duplicate record): %s', records[0]['id'])
+            return True
+
+        # Make sure TXT records are wrapped in quotes
+        if content:
+            content = self._add_quotes(rtype, content)
+
+        data = {
+            'name': self._full_name(name),
+            'type': rtype,
+            'content': content
+        }
+
+        self._post('/zones/{0}/records'.format(self.domain), data)
+        LOGGER.debug('create_record: %s', True)
+        return True
+
+    def _update_record(self, identifier, rtype=None, name=None, content=None):
+
+        # Make sure TXT records are wrapped in quotes
+        if content:
+            content = self._add_quotes(rtype, content)
+
+        data = {}
+        if name:
+            data['name'] = self._full_name(name)
+        if rtype:
+            data['type'] = rtype
+        if content:
+            data['content'] = content
+
+        self._patch(
+            '/zones/{0}/records/{1}'.format(self.domain, identifier), data)
+
+        LOGGER.debug('update_record: %s', True)
+        return True
+
+    def _delete_record(self, identifier=None, rtype=None, name=None, content=None):
+
+        delete_record_ids = []
+
+        # If we've not been given an identifier, search for matching records.
+        # NOTE, this could cause multiple records to be removed.
+        if not identifier:
+            records = self._list_records(rtype, name, content)
+            delete_record_ids = [record['id'] for record in records]
+        else:
+            delete_record_ids.append(identifier)
+
+        LOGGER.debug('delete_records: %s', delete_record_ids)
+
+        for delete_record_id in delete_record_ids:
+            self._delete(
+                '/zones/{0}/records/{1}'.format(self.domain, delete_record_id))
+
+        LOGGER.debug('delete_record: %s', True)
+        return True
+
+    # As there is no _patch method available from Provider, let's create our own
+    # by creating a _request with action 'PATCH'.
+    def _patch(self, url='/', data=None, query_params=None):
```

Comment:
I think you can move this method to the base provider, so everyone can benefit from it.

Reply:
Yep, good shout